privaterelay/settings.py

```diff
@@ -183,34 +184,42 @@
     empty_hash = base64.b64encode(sha256().digest()).decode()
     _CSP_STYLE_HASHES.append(f"'sha256-{empty_hash}'")
 
-CSP_DEFAULT_SRC = ["'self'"]
-CSP_CONNECT_SRC = [
-    "'self'",
-    "https://www.google-analytics.com/",
-    "https://location.services.mozilla.com",
-    "https://api.stripe.com",
-    BASKET_ORIGIN,
-] + _ACCOUNT_CONNECT_SRC
-CSP_FONT_SRC = ["'self'"] + _API_DOCS_CSP_FONT_SRC + ["https://relay.firefox.com/"]
-CSP_IMG_SRC = ["'self'"] + _AVATAR_IMG_SRC + _API_DOCS_CSP_IMG_SRC
-CSP_SCRIPT_SRC = (
-    ["'self'"]
-    + (["'unsafe-inline'"] if _CSP_SCRIPT_INLINE else [])
-    + [
-        "https://www.google-analytics.com/",
-        "https://js.stripe.com/",
-    ]
-)
-CSP_WORKER_SRC = _API_DOCS_CSP_WORKER_SRC or None
-CSP_OBJECT_SRC = ["'none'"]
-CSP_FRAME_SRC = ["https://js.stripe.com", "https://hooks.stripe.com"]
-CSP_STYLE_SRC = (
-    ["'self'"]
-    + (["'unsafe-inline'"] if _CSP_STYLE_INLINE else [])
-    + _API_DOCS_CSP_STYLE_SRC
-    + _CSP_STYLE_HASHES
-)
-CSP_REPORT_URI = config("CSP_REPORT_URI", "")
+
+CONTENT_SECURITY_POLICY: CONTENT_SECURITY_POLICY_T = {
+    "DIRECTIVES": {
+        "default-src": [SELF],
+        "connect-src": [
+            SELF,
+            "https://www.google-analytics.com/",
+            "https://location.services.mozilla.com",
+            "https://api.stripe.com",
+            BASKET_ORIGIN,
```

Comment:
question (non-blocking): Why do we include this constant `BASKET_ORIGIN` inline here, but then use `.extend()` to add the other `_ACCOUNT_CONNECT_SRC` constant?

Reply:
`BASKET_ORIGIN` is a single string in every environment, while the others are lists that change content and size (for example, 0 length in one environment, 2 in others).

I tried to make the changes small, since there is no way to validate a CSP outside of running it on a server. but we could certainly re-arrange the logic. The setup code above `CONTENT_SECURITY_POLICY` could move to conditionals below it.

Reply:
Nah, that's fine. So long as it was intentional.